test/foundry/deploy/15_holograph_registry_tests.t.sol

```diff
@@ -0,0 +1,802 @@
+        vm.expectRevert('HOLOGRAPH: contract already set');
+        registry.referenceContractTypeAddress(address(holographERC20));          
+    }
+
+/**
+ * @notice Test allowing an external contract to call the referenceContractTypeAddress function.
+ * @dev This test verifies that an external contract can successfully call the referenceContractTypeAddress function
+ * of the registry contract with the address of the holographERC20 contract as a parameter.
+ * Refers to the hardhat test with the description 'Should allow external contract to call fn'
+ */
+    function testAllowExternalContractToCallFnReferenceContractTypeAddress() public {
+        bytes memory encodeSignature = abi.encodeWithSignature('referenceContractTypeAddress(address)', address(holographERC20));
+        mockExternalCall.callExternalFn(address(registry), encodeSignature);        
+    }
+
+/*
+    setHolograph()
+*/   
+
+/**
+ * @notice Test allowing an admin to alter the holograph slot in the registry.
+ * @dev This test verifies that an admin can successfully update the holograph slot
+ * in the registry with a new address.
+ * It simulates the admin calling the setHolograph function with a mock address,
+ * and then checks if the registry correctly stores and returns the updated holograph address.
+ * Refers to the hardhat test with the description 'should allow admin to alter _holographSlot'
+ */
+    function testAllowAdminToAlter_holographSlot() public {
+        // TODO Check which is the admin adddres
+        vm.startPrank(holographRegistry.getAdmin());
+        holographRegistry.setHolograph(address(mockAddress));
+        assertEq(holographRegistry.getHolograph(), mockAddress);
+        vm.stopPrank();
+    }
+
+/**
+ * @notice Test the revert when the owner tries to alter the holograph slot in the registry.
+ * @dev This test verifies that the registry reverts with the message 'HOLOGRAPH: admin only function'
+ * when the owner (not the admin) attempts to alter the holograph slot.
+ * It simulates the owner trying to call the setHolograph function with a mock address.
+ * Refers to the hardhat test with the description 'should fail to allow owner to alter _holographSlot'
+ */
+    function testAllowOwnerToAlter_holographSlotFail() public {
+        vm.expectRevert('HOLOGRAPH: admin only function');
+        holographRegistry.setHolograph(address(mockAddress));
+    }
+
+/**
+ * @notice Test the revert when a random user tries to alter the holograph slot in the registry.
+ * @dev This test verifies that the registry reverts with the message 'HOLOGRAPH: admin only function'
+ * when a random user (not the admin) attempts to alter the holograph slot.
+ * It simulates a random user calling the setHolograph function with a mock address.
+ * Refers to the hardhat test with the description 'should fail to allow non-owner to alter _holographSlot'
+ */
+    function testAllowRandUserToAlter_holographSlotFail() public {
```

Comment:
how does it differ from the test above, if they are similar, I would prank a non-admin address and just leave that one.